.circleci/config.yml

```diff
@@ -0,0 +1,218 @@
+version: 2
+
+environment: &environment
+  - IBIS_TEST_POSTGRES_DB: ubuntu
+  - IBIS_POSTGRES_USER: ubuntu
+  - IBIS_POSTGRES_PASS: ubuntu
+
+  - DATA_URL: https://storage.googleapis.com/ibis-ci-data
+
+  # The following environment variables are necessary to run impala tests
+  - IBIS_TEST_IMPALA_HOST: localhost
+  - IBIS_TEST_IMPALA_PORT: 21050
+  - IBIS_TEST_NN_HOST: localhost
+  - IBIS_TEST_WEBHDFS_PORT: 50070
+  - IBIS_TEST_WEBHDFS_USER: ubuntu
+
+python_test_steps: &python_test_steps
+  - checkout
+  - attach_workspace:
+      at: ~/workspace
+  - run: cat "$HOME/workspace/envars.sh" >> $BASH_ENV
+  - run: echo 'export ENV_NAME=ibis_${PYTHON_VERSION}' >> $BASH_ENV
+  - run: sudo apt-get install -qq clang libboost-dev
+  - run: conda env create --quiet --name "${ENV_NAME}" --file "ci/requirements-dev-${PYTHON_VERSION}.yml"
+  - run: echo 'source activate ${ENV_NAME}' >> $BASH_ENV
+  - run: flake8
+  - run: python setup.py develop
+  - run: dockerize -wait tcp://localhost:5432 -wait tcp://localhost:21050 -wait tcp://localhost:50070 -timeout 1m
+  - run: test_data_admin.py load --data --data-dir "$HOME/workspace/ibis-testing-data"
+  - run: pytest -m 'not impala and not hdfs' --doctest-modules --doctest-ignore-import-errors -rsxX --tb=short ibis
+  - run: pytest -m 'impala or hdfs' --doctest-modules --doctest-ignore-import-errors -rsxX --tb=short ibis --durations=20
```

Comment:
One benefit is that each step is run in isolation on circle ci, so you can see exactly which command failed and you can look at only that command's output and error messages. With a shell script you have to some hunting. I think one other (possibly only marginal) benefit is that all of the steps are visible in the circle yaml, which means I only have to have one file open to know what's going on.